src/DirectoryManager.cpp

```diff
@@ -0,0 +1,59 @@
+#include "DirectoryManager.h"
+#include <iostream>
+#include <sstream>
+#include <unistd.h>
+#include <sys/wait.h>
+
+
+/*
+* get the current directory.
+* @return the current directory as string.
+*/
+std::string DirectoryManager::getCurrentDirectory()
+{
+    char cwd[PATH_MAX];
+    return (getcwd(cwd, sizeof(cwd)) != nullptr) ?
+        std::string(cwd) : ERROR_UNKNOWN_DIR;
+}
+
+
+/*
+* change the current directory path.
+* @param args - the command and its arguments.
+*/
+void DirectoryManager::changeDirectory(const std::string& newPath)
+{
+    if (newPath.empty()) {
+        throw std::runtime_error(ERROR_UNSPECIFIED_DIR);
+    }
+    if (chdir(newPath.c_str()) != 0) {
```

Comment:
again, using std::filesystem is the modern c++ way: https://en.cppreference.com/w/cpp/filesystem/current_path